go-controller/pkg/node/gateway_iptables.go

```diff
@@ -336,13 +370,14 @@ func recreateIPTRules(table, chain string, keepIPTRules []iptRule) {
 	}
 }
 
-// getGatewayIPTRules returns NodePort and ExternalIP iptables rules for service. If nodeIP is non-nil, then
-// only incoming traffic on that IP will be accepted for NodePort rules; otherwise incoming traffic on the NodePort
-// on all IPs will be accepted. If gatewayIP is "", then NodePort traffic will be DNAT'ed to the service port on
-// the service's ClusterIP. Otherwise, it will be DNAT'ed to the NodePort on the gatewayIP.
-func getGatewayIPTRules(service *kapi.Service, hasLocalHostEndpoint bool) []iptRule {
+// getGatewayIPTRules returns NodePort and ExternalIP iptables rules for service.
+// If svcHasLocalHostNetEndPnt and svcTypeIsETPLocal, rule that redirects traffic to host targetPort is added.
+// If svcHasNonLocalHostNetEndPnt and svcTypeIsETPLocal in LGW mode, rule that redirects traffic to ovn-k8s-mp0 is added.
+// In all other cases, DNAT rule towards clusterIP svc is added.
+func getGatewayIPTRules(service *kapi.Service, svcHasLocalHostNetEndPnt, svcHasNonLocalHostNetEndPnt bool) []iptRule {
```

Comment:
Is it necessary to have both bools like this?
svcHasLocalHostNetEndPnt, svcHasNonLocalHostNetEndPnt seem like opposites and you could just reduce to one? could just be localHostNetworkEndpoint

Local means local to the node, and not local gw right? Might be good to add that in the comment.

Reply:
I think i realize now that svcHasNonLocalHostNetEndPnt really means has OVN networked local endpoints and we should change the variable name to be more clear.

Reply:
> I think i realize now that svcHasNonLocalHostNetEndPnt really means has OVN networked local endpoints and we should change the variable name to be more clear.

No Tim, `svcHasNonLocalHostNetEndPnt` means the endpoints are not - local-to-node-host-networked-endpoints :)
It does not imply "it has OVN networked local endpoints", since it also includes cases like ETP=local+host-networked-endpoints-on-different-nodes. I know this is confusing...